src/main/java/seedu/tp/ui/Ui.java

```diff
@@ -75,6 +80,55 @@ public String promptUserForRequiredField(String fieldName) {
         return input;
     }
 
+    /**
+     * Prompts the user for an optional date used in the construction of a <code>Flashcard</code>.
+     *
+     * @param fieldName string representing name of the date to prompt for
+     * @return the parsed date
+     */
+    public Optional<LocalDate> promptUserForOptionalLocalDate(String fieldName) {
```

Comment:
If the date is optional, why do we still prompt user to enter again when the input is empty?

Reply:
When the input is empty it returns Optional.empty().
When the input is non-empty is tries to parse a date. If the date isn't parseable then it will ask again

Reply:
Understood. Sorry I missed the `EMPTY_STRING` part!